zerver/actions/message_edit.py

```diff
@@ -1409,9 +1448,10 @@ def check_update_message(
         new_stream = access_stream_by_id(user_profile, stream_id, require_active=True)[0]
         check_stream_access_based_on_stream_post_policy(user_profile, new_stream)
 
-    check_time_limit_for_stream_or_topic_changes(
-        message, user_profile, stream_id, topic_name, propagate_mode, edit_limit_buffer
-    )
+    if not is_resolve_topic_request(message, topic_name, propagate_mode, content):
+        check_time_limit_for_stream_or_topic_changes(
```

Comment:
I guess this will not check the stream moving time limit even when stream is being changed if the topic is being updated by adding the checkmark as prefix.

Reply:
According to the issue description,
> There should not be an option to time-limit topic resolution.

So, I'm not sure whether that check should be performed.

Reply:
But if the stream is being changed, then time limit for changing stream should be checked.

Reply:
updated!